packages/core/src/entity/EntityDefinition.ts

```diff
@@ -143,6 +143,17 @@ export class EntityDefinition {
 						const extractedJsonSchema = transformEntityDefinitionSchema(
 							omit(entityPropDecorator?.options ?? {}, ['type', 'required']),
 							args => {
+								const enm = args.schema.enum;
+								if (enm && typeof enm === 'object') {
+									if (type.name === 'Number') {
+										args.schema.enum = Object.values(enm)
```

Comment:
It's better to perform the transformations in an immutable way, without mutating the arguments

Reply:
Hey @pierissimo. I made some changes to support complex types like (anyOf, etc.) as discussed in slack. But still not sure how to avoid mutating the args

Reply:
@adriasala124 I've slightly modified your code to make the enum manipulations immutable, as discussed before

Reply:
Thank you @pierissimo I just checked it. If you give me approval I will merge if you think it's good.